### PR TITLE
Pretty much just fix and redo a lot of stuff

### PR DESCRIPTION
**Pull request by tandemdude**
See commit messages

### CODE REVIEW

File: modules/loader.py [hunk 1]
```diff
@@ -6,6 +6,21 @@ def __init__(self, bot):
...
 
+    async def cog_command_error(self, ctx, error):
+        if isinstance(error, commands.errors.CommandInvokeError):
+            print(f"[WARN] {str(error)}")
+            if isinstance(error.original, commands.errors.ExtensionNotLoaded):
+                return await ctx.send("[!] Module not already loaded.")
+            elif isinstance(error.original, commands.errors.ExtensionAlreadyLoaded):
+                return await ctx.send("[!] Module already loaded.")
+            elif isinstance(error.original, commands.errors.ExtensionNotFound):
+                print("YEEET")
```
Comment (the-emerald) [inline modules/loader.py]: Please remove this. (It's funny though)

Comment (tandemdude) [reply modules/loader.py]: Whoops that was debug I forgot to remove lol

**Comment by the-emerald** [comment]
Changelist and approval:
- [x] Change role handling
- [x] Use listener for logins
- [x] Use listener for error handling
- [x] Add custom help command
- [ ] ~~Add unit conversion~~
- [x] Add support for versionless cog edge case
- [x] Various QoL changes

**Comment by the-emerald** [comment]
~~**Add support for versionless cog edge case:**
Not a big fan of the current behaviour of skipping a cog entirely when the version isn't well defined (perhaps make it go `(?.?.?)`)?~~

**Comment by the-emerald** [comment]
~~**Add custom help command:**
I'll have to write details for every command if this change is approved.~~

~~Perhaps group commands by module? The demo is fine, but it could get messy with more modules are loaded.~~

**Comment by tandemdude** [comment]
> **Add custom help command:**
> I'll have to write details for every command if this change is approved.
> 
> Perhaps group commands by module? The demo is fine, but it could get messy with more modules are loaded.

I'm currently adding details to all the commands I have edited in some way. Grouping by cog/module can be done - will work on that now

**Comment by the-emerald** [comment]
**Add unit conversion:**
We were originally thinking about deprecating this module and substituting it with something like WolframAlpha. @QuickRecon what do you think?

**Comment by the-emerald** [comment]
**Use listener for logins** and
**Use listener for error handling:**

Your rationale is well thought out, but I'll have to run it by the other contributors first before I can accept it. @QuickRecon @hxtk please take a look.

File: modules/status.py [hunk 2]
```diff
@@ -9,8 +9,7 @@ def __init__(self, bot):
         self.bot = bot
         self.start_time = datetime.datetime.utcnow()
 
-    @staticmethod
-    def uptime_convert(seconds):
+    def uptime_convert(self, seconds):
```
Comment (the-emerald) [inline modules/status.py]: Should be static, as discussed

Comment (tandemdude) [reply modules/status.py]: Already changed on my machine

File: bot.py [hunk 3]
```diff
@@ -1,76 +1,61 @@
...
+                bot.load_extension(module)
+                print(f"[LOAD] Successfully loaded extension {module}")
+            except Exception as e:
+                print(e)
+                print(f"[ERROR] Failed to load extension {module}, continuing.")
+    bot.run(token)
+
+
+if __name__ == '__main__':
+    while True:
```
Comment (the-emerald) [inline bot.py]: Is there a better way of implementing this? Not a big fan of a `while True` loop.

File: modules/units.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import discord
```
Comment (the-emerald) [inline modules/units.py]: Please use spaces instead of tabs across this file, in line with the rest of the code base.

File: modules/listeners.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import discord
+from discord.ext import commands
+import traceback
+import sys
+
+
+class Listeners(commands.Cog):
```
Comment (the-emerald) [inline modules/listeners.py:7]: I like this, but I'll have to implement Cog visibilities in the `Status` module before I can approve this module.

**Review by the-emerald — CHANGES_REQUESTED** [review]

File: bot.py (hunk 3 above)
Comment (the-emerald) [reply bot.py]: After reading and asking around I really couldn't find a single discord.py who found this necessary. I don't think it's worth the extra trouble.

**Review by the-emerald — CHANGES_REQUESTED** [review]
Everything looks good. However I'll be implementing a wolfram-alpha module soon so `units` will be redundant.

Unfortunately I still have to implement cog visibilities before I can approve the listeners refactor. If you want to you can set it up like this: make a `visible: bool` field for every module, and then check if it's a false (perhaps default to True otherwise).

File: modules/units.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+from discord.ext import commands
```
Comment (the-emerald) [inline modules/units.py]: Remove this module

**Review by the-emerald — APPROVED** [review]

**Pull request merged**